backends/platform/atari/readme.txt

```diff
@@ -0,0 +1,414 @@
+synchronizes the point when the rendering process begins - if it takes more
+than the time reserved for the vertical blank interrupt (what happens
+with most of the games), you'll see screen tearing.
+
+Pros:
+
+- fastest possible rendering (especially in 640x480 with a lot of small
+  rectangle updates where the buffer copying drags performance down)
+
+Cons:
+
+- screen tearing in most cases
+
+- SuperVidel only: using C2P would be not only suboptimal (every rectangle
+  would be C2P'ed instead of just copy and C2P of the final screen) but poses an
+  additional problem as C2P requires data aligned on a 16px boundary and
+  ScummVM supplies arbitrarily-sized rectangles (this is solvable by custom
+  Surface allocation but it's not bullet-proof). In theory I could implement
+  direct rendering for the Falcon hicolor (320x240@16bpp) but this creates
+  another set of issues like when palette would be updated but not the whole
+  screen - so some rectangles would be rendered in old palette and some in new.
+
+SuperBlitter used: sometimes (when ScummVM allocates surface via its create()
+function; custom/small buffers originating in the engine code are still copied
+using the CPU).
+
+Single buffering:
+~~~~~~~~~~~~~~~~~
+
+This is very similar to the previous mode with the difference that the engine
+uses an intermediate buffer for storing the rectangles but yet it remembers
+which ones they were. It works also on plain Videl and applies the chunky to
+planar process to each one of the rectangles separately, avoiding fullscreen
+updates (but if such is needed, there is an optimized code path for it). Vsync()
+is used the same way as in the previous mode, i.e. screen tearing is still
+possible.
+
+Pros:
+
+- second fastest possible rendering
+
+- doesn't update the whole screen (works best with a moderate amount of
+  rectangles to update)
+
+Cons:
+
+- screen tearing in most cases
+
+- if there are too many smaller rectangles, it can be less efficient than
+  updating the whole buffer at once
+
+SuperBlitter used: yes, for rectangle blitting to screen and cursor restoration.
+Sometimes also for generic copying between buffers (see above).
+
+Double buffering:
+~~~~~~~~~~~~~~~~~
+
+The most common rendering mode. It extends the idea of single buffering - it
+renders into two buffers, one is visible while the other one is used for
+updating. At the end of the update process the two buffers are swapped, so the
+newly updated one is displayed. By definition, Vsync() must be always enabled
+(the buffers are swapped in the vertical blank handler) otherwise you'd see
+screen tearing.
+
+Pros:
+
+- stable frame rate, leading to fixed e.g. 30 FPS rendering for the whole time
+  if game takes, say, 1.7 - 1.9 frames per update
+
+- no screen tearing in any situation
+
+Cons:
+
+- since two buffers are present, the buffer is always blitted into the screen
+  surface as whole, even if only one tiny little rectangle is changed (excluding
+  the cursor)
```

Comment:
The SDL backend combines the dirty rectangles for the current frame with the ones for the previous frame when double buffering is enabled. Would doing something like that help here as well?

Reply:
What a great idea! Added as another future enhancement. :)